osf/management/commands/international_user_metrics.py

```diff
@@ -0,0 +1,181 @@
+    children_json = resp.json()['data']
+    try:
+        return next(x['links']['upload'] for x in children_json if x['attributes']['name'] == filename)
+    except StopIteration:
+        resp = requests.put(
+            base_folder_data['links']['upload'] + '?name={}&kind=file'.format(filename),
+            data=data,
+            headers=headers
+        )
+        resp.raise_for_status()
+        return resp.json()['data']['links']['upload']
+
+
+def create_zip(zip_name, file_names, files):
+    buff = BytesIO()
+    with zipfile.ZipFile(buff, mode='w') as zip_file:
+        zip_file.filename = zip_name
+        for i, data in enumerate(files):
+            zip_file.writestr(file_names[i], data)
+
+    buff.seek(0)
+    return buff.read()
+
+
+def create_csv(rows):
+    file = BytesIO()
+    writer = csv.DictWriter(file, fieldnames=list(rows[0].keys()))
+    writer.writeheader()
+    for row in rows:
+        writer.writerow(row)
+
+    file.seek(0)
+    return file.read()
+
+
+def update_international_users_counts():
+    # International users aggregate counts
+    new_row = [{
+        'date': timezone.now().strftime('%Y-%m-%d'),
+        'users_usa_storage': get_user_count_by_region(USA),
+        'users_germany_storage': get_user_count_by_region(GERMANY),
+        'users_canada_storage': get_user_count_by_region(CANADA),
+        'users_australia_storage': get_user_count_by_region(AUSTRALIA),
+    }]
+
+    fieldnames = list(new_row[0].keys())
+
+    # Check if file exists add header if so.
+    upload_link = get_or_create_file(
+        DS_METRICS_BASE_FOLDER,
+        FILENAME,
+        DS_METRICS_OSF_TOKEN,
+        create_csv_header(fieldnames)
+    )
+
+    # Now it definitely exists, download the old add the new and upload.
+    append_to_csv(upload_link, new_row, DS_METRICS_OSF_TOKEN)
+
+
+def upload_users_raw_data():
+    """ This creates and uploads a zip file containing CSVs with relevant user data."""
+    # Raw user data
+    rows = OSFUser.objects.annotate(
+        # Rename region_id for less awkward column title
+        region_id=F('addons_osfstorage_user_settings__default_region_id')
+    ).values('username', 'id', 'region_id')
+
+    chunks = [rows[i:i + USER_PER_CSV] for i in range(0, len(rows), USER_PER_CSV)]
+
+    filenames = []
+    files = []
+    for i, chunk in enumerate(chunks):
+        files.append(create_csv(chunk))
+        filenames.append('data-usage-raw-user-{}.csv'.format(i))
+
+    zip_filename = 'data_storage_raw_users_{}.zip'.format(datetime.datetime.now().strftime('%Y-%m-%d'))
+    zip_data = create_zip(zip_filename, filenames, files)
+
+    upload_link = get_or_create_file(DS_METRICS_BASE_FOLDER, zip_filename, DS_METRICS_OSF_TOKEN, zip_data)
+    upload_to_file(upload_link, zip_data, DS_METRICS_OSF_TOKEN)
+
+
+@celery_app.task(name='management.commands.international_user_metrics')
+def main():
+    update_international_users_counts()
+    upload_users_raw_data()
+
+
+class Command(BaseCommand):
```

Comment:
This should have a DRY_RUN option that will run all of the queries and output the summary data but will not write to any files nor try to upload anything.